docs/ldml/tr35-personNames.md

```diff
@@ -41,26 +41,929 @@ The LDML specification is divided into the following parts:
+
+### 6.1 <a name="6-1-derive-the-name-locale" href="#6-1-derive-the-name-locale">Derive the name locale</a>
+
+If there is no name locale available via the PersonName data interface, then an effective name locale is created based on the characters in the name and the formatting locale. 
+
+* Get the predominant script of the given and surname, and find the likely language as specified by the likely subtags. Use the locale form from the likely language and the script as the name locale.
+
+### 6.2 <a name="6-2-derive-the-formatting-locale" href="#6-2-derive-the-formatting-locale">Derive the formatting locale</a>
+
+If the name locale is different from the formatting locale, and the script of the name is different from the script of the formatting locale, then change the formatting locale to the name locale.
+
+### 6.3 <a name="6-3-derive-the-name-order" href="#6-3-derive-the-name-order">Derive the name order</a>
+
+The name order data provides a mapping from the name locale to an order parameter. In current data, the important part of the name locale is the region code, but allowing a locale code provides for easy extension in the future.`  `
+
+A PersonName object’s fields are used to derive an order, as follows:
+
+1. If the PersonName object to be formatted has a preferredOrder_ _field, then return that field’s value
+2. Otherwise, if the PersonName object has a nameLocale field, and there is a nameOrderLocales element:
+    1. Map the nameLocale field using the likelySubtags data, eg zh ⇒ zh_Hans_CN
+    2. Traverse the nameOrderLocales elements, and within each order attribute value to see if there is a match. 
+    3. As usual, a **und** (undefined) language subtag or empty script or region in the order attribute locale matches any language subtag, script subtag, or region subtag (respectively) in the PersonName object’s locale field.
+    4. If there is a match, return the nameOrder elements order attribute value.
+3. Otherwise, return givenFirst.
+
+For example, the data for a particular locale might look like the following:
+
+```xml
+<nameOrderLocales order="surnameFirst">zh ja und-CN und-TW und-SG und-HK und-MO und-HU und-JP</nameOrderLocales>
+```
+
+The nameOrderLocales will match any locale with a zh or ja [unicode_language_subtag](https://unicode.org/reports/tr35/#unicode_language_subtag) and any locale with a CN, TW, SG, HK MO, HU, or JP [unicode_region_subtag](https://unicode.org/reports/tr35/#unicode_region_subtag).
+
+Here are some more examples. Note that if there is no order field or locale field in the PersonName object to be formatted, and the script of the PersonName data is different from that of the formatting locale, then the default result is givenFirst.
+
+| Name Object preferredOrder | Name Object locale | Resulting Order |
+| -------------------------- | ------------------ | --------------- |
+| surnameFirst               |                    | surnameFirst    |
```

Comment:
for the name object locale in this line, maybe put <any>?

Reply:
Done